mirgecom/filter.py

```diff
@@ -201,7 +201,13 @@ def filter_modally(dcoll, dd, cutoff, mode_resp_func, field):
     dd_modal = dof_desc.DD_VOLUME_MODAL
     discr = dcoll.discr_from_dd(dd)
 
-    assert isinstance(field, DOFArray)
+    from arraycontext import map_array_container
+    from functools import partial
+    if not isinstance(field, DOFArray):
+        return map_array_container(
+            partial(filter_modally, dcoll, dd, cutoff, mode_resp_func), field
+        )
+
```

Comment:
You're totally right that `@obj_array_vectorize_n_args` can be removed, thanks for pointing that out. I also updated the docs to reflect the changes: https://github.com/illinois-ceesd/mirgecom/pull/566/commits/74558fa31d5f7a4807117a4e4c527b7e6fceef5a

> I think with [inducer/arraycontext#128](https://github.com/inducer/arraycontext/pull/128) it may be possible to implement this function as something along the lines of:
> 
> ```python
>     def filter_scalar(f):
>         dd_nodal = dof_desc.as_dofdesc(dd)
>         dd_modal = dof_desc.DD_VOLUME_MODAL
>         discr = dcoll.discr_from_dd(dd_nodal)
>         actx = f.array_context
>         modal_map = dcoll.connection_from_dds(dd_nodal, dd_modal)
>         nodal_map = dcoll.connection_from_dds(dd_modal, dd_nodal)
>         f = modal_map(f)
>         f = apply_spectral_filter(actx, f, discr, cutoff, mode_resp_func)
>         return nodal_map(f)
> 
>     from arraycontext import rec_map_array_container
>     return rec_map_array_container(filter_scalar, field, leaf_class=DOFArray)
> ```

I think it's certainly an option. However, I'd prefer not to tackle that in this PR.